oauth2.cgi

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env perl
```

Comment:
shouldn't this be a mojo route rather than a cgi?

Reply:
That was my original plan. But after trying various hacks to make it work I gave in and made it a small cgi. Reason is that the way our current Auth stack works relies heavily on running inside the CGI compatibility wrapper that we use with Mojolicious. I would either need to do quite a bit of refactoring or duplicating a lot of code to make it work. 

My future plan is to start moving most of the supporting code (login, utilities, etc.) into native Mojo and then doing these things will be much less painful.